lib/_stream_readable.js

```diff
@@ -660,7 +661,7 @@ Readable.prototype.on = function(ev, fn) {
     this.resume();
   }
 
-  if (ev === 'readable' && this.readable) {
+  if (ev === 'readable') {
```

Comment:
Now that `.readable` can possibly be false, the whole conditional has a chance of evaluating to false, which causes test failures. Since the second condition was always evaluating to true before this, it can be safely removed and all tests pass again.